homeassistant/components/fritzbox/sensor.py

```diff
@@ -62,6 +63,21 @@ async def async_setup_entry(
                     ain,
                 )
             )
+            # Only some temperature devices are also humidity sensors
+            if device.rel_humidity is not None:
+                entities.append(
+                    FritzBoxHumiditySensor(
+                        {
+                            ATTR_NAME: f"{device.name} Humidity",
+                            ATTR_ENTITY_ID: f"{device.ain}_humidity",
+                            ATTR_UNIT_OF_MEASUREMENT: PERCENTAGE,
+                            ATTR_DEVICE_CLASS: DEVICE_CLASS_HUMIDITY,
+                            ATTR_STATE_CLASS: STATE_CLASS_MEASUREMENT,
+                        },
+                        coordinator,
+                        ain,
+                    )
+                )
```

Comment:
please change the indention, so that is is on same level as `if device.has_temperature_sensor`
```suggestion

        if device.rel_humidity is not None:
            entities.append(
                FritzBoxHumiditySensor(
                    {
                        ATTR_NAME: f"{device.name} Humidity",
                        ATTR_ENTITY_ID: f"{device.ain}_humidity",
                        ATTR_UNIT_OF_MEASUREMENT: PERCENTAGE,
                        ATTR_DEVICE_CLASS: DEVICE_CLASS_HUMIDITY,
                        ATTR_STATE_CLASS: STATE_CLASS_MEASUREMENT,
                    },
                    coordinator,
                    ain,
                )
            )
```

Reply:
`rel_humidity` is a field of `FritzhomeDeviceTemperature`, so it makes sense to check for `has_temperature_sensor`, doesn't it?

Reply:
that's correct, but the `pyfritzhome` does always publish the cumulated `FritzhomeDevice` class, those attribute `rel_humidity` (_also all others_) are always available on `device`. The more specific classes like `FritzhomeDeviceTemperature` or `FritzhomeDeviceThermostat` are only internal used to get the lib better structured.
Furthermore with the suggestion `FritzBoxHumiditySensor` can still be created, even it is part of an `FritzhomeDeviceThermostat`